SingularityService/src/main/java/com/hubspot/singularity/data/DeployManager.java

```diff
@@ -99,6 +103,13 @@ public DeployManager(
     this.updateRequestTranscoder = updateRequestTranscoder;
     this.deploysCache = deploysCache;
     this.leaderCache = leaderCache;
+    this.deployCache =
+      new ApiCache<>(
+        configuration.useApiCacheInDeployManager(),
+        configuration.getDeployCacheTtl(),
+        this::fetchAllDeployStates,
+        executorServiceFactory.get("deploy-api-cache-reloader")
```

Comment:
I _think_ there is a version of this that returns a single threaded executor, which is all we should need for this case

Reply:
We don't have an option for a single threaded ScheduledExecutorService in SingularityManagedScheduledExecutorServiceFactory but Executors has newSingleThreadScheduledExecutor that I can add.